detail.py

```diff
@@ -0,0 +1,25 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from catalog import models as cmod
+
+@view_function
+def process_request(request, prod:cmod.Product):
+
+    for index, item in enumerate(request.last_five):
+        if item.id == prod.id:
+            request.last_five.pop(index)
+        elif len(request.last_five) > 6:
+            request.last_five.pop
+
+    request.last_five.insert(0, prod)
+
+    p1 = cmod.Product.objects.get(id = request.dmp.urlparams[0])
```

Comment:
Instead of urlparams[0], you might think about just accessing your 'prod' parameter in the process_request function

Reply:
Ya I realized that too because I already bring in prod.